infra/enrolment-records-store/src/main/java/com/simprints/infra/enrolment/records/store/commcare/CommCareIdentityDataSource.kt

```diff
@@ -0,0 +1,163 @@
+        FingerprintIdentity(
+            it.payload.subjectId,
+            it.payload.biometricReferences.filterIsInstance<FingerprintReference>()
+            .flatMap { fingerprintReference ->
+                fingerprintReference.templates.map { fingerprintTemplate ->
+                    FingerprintSample(
+                        fingerIdentifier = fingerprintTemplate.finger,
+                        templateQualityScore = fingerprintTemplate.quality,
+                        template = encoder.base64ToBytes(fingerprintTemplate.template),
+                        format = fingerprintReference.format,
+                    )
+                }
+            }
+        )
+    }
+
+    override suspend fun loadFaceIdentities(
+        query: SubjectQuery,
+        range: IntRange,
+    ): List<FaceIdentity> = loadEnrolmentRecordCreationEvents(range).map {
+        FaceIdentity(
+            it.payload.subjectId,
+            it.payload.biometricReferences.filterIsInstance<FaceReference>()
+            .flatMap { faceReference ->
+                faceReference.templates.map { faceTemplate ->
+                    FaceSample(
+                        template = encoder.base64ToBytes(faceTemplate.template),
+                        format = faceReference.format,
+                    )
+                }
+            }
+        )
+    }
+
+    private fun loadEnrolmentRecordCreationEvents(range: IntRange): List<EnrolmentRecordCreationEvent> {
+        val enrolmentRecordCreationEvents: MutableList<EnrolmentRecordCreationEvent> = mutableListOf()
+
+        try {
+            val caseMetadataCursor =
+                context.contentResolver.query(CASE_METADATA_URI, null, null, null, null)
+
+            if (caseMetadataCursor != null && caseMetadataCursor.moveToPosition(range.first)) {
+                do {
+                    val caseId =  caseMetadataCursor.getString(caseMetadataCursor.getColumnIndexOrThrow(COLUMN_CASE_ID))
+
+                    caseId?.let {
+                        enrolmentRecordCreationEvents.addAll(loadEnrolmentRecordCreationEvents(it))
+                    }
+
+                } while (caseMetadataCursor.moveToNext() && caseMetadataCursor.position < range.last)
+            }
+
+            caseMetadataCursor?.close()
+        } catch (e: Exception) {
+            Simber.e("Error while querying CommCare", e)
+        }
+
+        return enrolmentRecordCreationEvents
+    }
+
+    private fun loadEnrolmentRecordCreationEvents(caseId: String): List<EnrolmentRecordCreationEvent> {
+        val caseEnrolmentRecordCreationEvents: MutableList<EnrolmentRecordCreationEvent> = mutableListOf()
+
+        //Access Case Data Listing for the caseId
+        val caseDataUri = CASE_DATA_URI.buildUpon().appendPath(caseId).build()
+        val caseDataCursor =
+            context.contentResolver.query(caseDataUri, null, null, null, null)
+        var subjectActions = ""
```

Comment:
I'd also utilized the [Kotlin's use](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/) inline function here